tests/validator/test_metrics_calculator.py

```diff
@@ -125,3 +126,47 @@ def test_column_partition_metric(
         )
         for idx, element in enumerate(results[desired_metric.id])
     )
+
+
+@pytest.mark.unit
+def test_get_metric_calls_get_metrics_and_returns_correct_result():
+    """
+    This basic test insures that MetricsCalculator.get_metric() uses MetricsCalculator.get_metrics() correctly.
+
+    In more detail, the purpose of this basic test is to insure that:
+        1) MetricsCalculator.get_metric() calls MetricsCalculator.get_metrics() exactly once for specific "metric_name";
+        2) MetricsCalculator.get_metric() correctly retrieves result from dictionary, returned by
+           MetricsCalculator.get_metrics() by using the specific "metric_name", mentioned above, as the key.
+
+    In the present test case, the role of "ExecutionEngine" is limited to providing the required constructor argument to
+    the "MetricsCalculator" class (one of whose methods is under test); hence, a "DummyExecutionEngine" is employed.
+
+    The "with mock.patch" is used judiciously, trading off the focus on the functionality under test (i.e., avoiding
+    "test leakage") against going as far as mocking all non-essential methods and properties, favoring code readability.
+    """
+
+    # noinspection PyAbstractClass
+    class DummyExecutionEngine(ExecutionEngine):
```

Comment:
```suggestion
    class DummyExecutionEngine:
```

You don't need to inherit here.

Reply:
@Kilo59 Then we will need cast.

Reply:
You don't really need to cast either, but If you want to do it go ahead.

Reply:
@Kilo59 Otherwise, we get PyCharm warnings (and when these warnings are justified, I feel it is better to accommodate them).